test.js

```diff
@@ -69,6 +70,29 @@ test('stripProtocol option', t => {
 	t.is(normalizeUrl('sindre://www.sorhus.com', options), 'sindre://sorhus.com');
 });
 
+test('stripTextFragment option', t => {
+	t.is(normalizeUrl('http://sindresorhus.com'), 'http://sindresorhus.com');
+	t.is(normalizeUrl('http://sindresorhus.com/about#'), 'http://sindresorhus.com/about');
+	t.is(normalizeUrl('http://sindresorhus.com/about#:~:text=hello'), 'http://sindresorhus.com/about');
```

Comment:
It should not strip the hash in this case. It's safe to strip a lone `#`, but here you lose information.

Reply:
I've made that change. I left the `stripHash` behaviour the same.

---

For my use case, I'll need to remove empty hashes manually afterwards then.

Background: I'm storing info about URLs in an extension and will be looking up the info on revisit (using the normalized URL again). To me, a URL with and without an empty hash should be treated as the same URL. I don't use `stripHash` because some sites might use it for routing. I doubt I'll run into an issue where a server will return a different result when there is an empty hash vs no empty hash.

This case of empty hashes after using `stripTextFragment` *will certainly* come up because I add a text fragment to the URLs presented to the user (which they use to revisit). It seems to be me that when a hash only exists for the sake of a text fragment, it's OK to strip it.